src/spikeinterface/extractors/sinapsrecordingh5extractor.py

```diff
@@ -0,0 +1,149 @@
+from pathlib import Path
+import numpy as np
+
+from probeinterface import get_probe
+
+from ..core.core_tools import define_function_from_class
+from ..core import BaseRecording, BaseRecordingSegment
+from ..preprocessing import UnsignedToSignedRecording
+
+
+class SinapsResearchPlatformH5RecordingExtractor_Unsigned(BaseRecording):
```

Comment:
Maybe we can just apply the unsigned_to_signed in the get_traces, which is simpler to read and will get rid of the complex class interplay. I can implement that

Reply:
done